libminifi/include/io/CRCStream.h

```diff
@@ -194,15 +194,12 @@ int CRCStream<T>::readData(uint8_t *buf, int buflen) {
 }
 
 template<typename T>
-int CRCStream<T>::writeData(std::vector<uint8_t> &buf, int buflen) {
-
-  if ((int)buf.capacity() < buflen)
```

Comment:
This existed for a reason. Did you test this in flight ? This causes a failure. We have non const because we reserve the right to adjust buffers. That's the intent behind making the signatures the way they are.